ui/src/ducks/inventory/index.js

```diff
@@ -1,17 +1,90 @@
+import 'axios' from axios
+import '../alerts/index.js'
 import { createAction, handleActions } from 'redux-actions'
 
 const actions = {
   INVENTORY_GET_ALL: 'inventory/get_all',
+  INVENTORY_GET_PENDING: 'inventory/get_all_pending',
+  INVENTORY_SAVE: 'inventory/save',
+  INVENTORY_DELETE: 'inventory/delete',
+  INVENTORY_REFRESH: 'inventory/refresh'
 }
 
 export let defaultState = {
-  all: []
+  all: [],
+  fetched: false,
 }
 
 export const findInventory = createAction(actions.INVENTORY_GET_ALL, () => {
-  //TODO
-})
+  (dispatch, getState, config) => axios
+    .get(`${config.restAPIUrl}/inventory`)
+    .then((suc) => {
+        dispatch(openSuccess)
+        dispatch(refreshInventory(suc.data))
+    })
+    }
+)}
+
+export const saveInventory = createAction(actions.INVENTORY_SAVE, (inventory) =>
+  (dispatch, getState, config) => axios
+    .post(`${config.restAPIUrl}/inventory`, inventory)
+    .then((suc) => {
+      const invs = []
+      getState().inventory.all.forEach(inv => {
+        if (inv.id !== suc.data.id) {
+          invs.push(inv)
+        }
+      })
+      invs.push(suc.data)
+      dispatch(openSuccess)
```

Comment:
openSuccess is a function. It will be called similarly to refreshInventory(). Also, openSuccess should be dispatched after refreshInventory (so we don't show a success message before the operation is fully complete)